drivers/serial/uart_gecko.c

```diff
@@ -669,7 +669,7 @@ static int uart_gecko_init(const struct device *dev)
 	return 0;
 }
 
-#ifdef CONFIG_PM_DEVICE
+#if defined(CONFIG_PM_DEVICE) && DT_HAS_COMPAT_STATUS_OKAY(silabs_gecko_usart)
```

Comment:
IMO it would be more correct to introduce `PM_DEVICE_DT_INST_DEFINE(idx, uart_gecko_pm_action);` in `GECKO_UART_INIT`.

Reply:
Sure, we can do that as well. I assumed the lack of support was intentional.

Reply:
Fixed

Reply:
And now also with the `PM_DEVICE_DT_INST_GET(idx)` call